packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -46,20 +46,37 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   componentRef?: IRefObject<ITextField>;
 
+  /**
+   * If true, the text field will follow strict controlled vs. uncontrolled behavior, similar to
+   * React's built-in input element:
+   * - If `value` is provided, `onChange` must be provided (unless `readOnly` is set to true) and
+   *   must manually update the component's props in response to user input. User input alone
+   *   will not update the displayed value.
+   * - Updates to `defaultValue` are not respected.
+   * - Changing the field from controlled (`value` provided) to uncontrolled (`defaultValue`
+   *   provided) is not supported.
+   * - Changing the field from uncontrolled to controlled is allowed but causes a warning.
+   *
+   * For more info, see https://reactjs.org/docs/forms.html and https://reactjs.org/docs/uncontrolled-components.html .
+   *
+   * In v7, this property will be removed and the behavior when it's set to true will become the default.
+   */
+  strictMode?: boolean;
```

Comment:
I am a bit sad by this one because this will almost guaranteed not be used. And if it does get used, it will need to be cleaned up later. But I understand not breaking people.

Reply:
This one would be easy to clean up with a codemod at least. :) I'm open to ideas for other strategies, but this was the best @JasonGore and I could think of to add the "right" behavior in a way that wouldn't dramatically break people in a non-major bump.